mca/mca.py

```diff
@@ -486,3 +486,76 @@ def _run_mca(self) -> None:
                             nh = self.topology.find_node(n)
                             nh.per_packet_traffic = True
                         self.explore_per_packet_diamond(node, ttl)
+
+    def send_extended_classification_probe(self,
+                                           flow_id_indices: tuple[int],
+                                           extended_classification_flow_id_index: int,
+                                           ttl: int):
+        """
+        Send one TTL limited extended classification probe.
+        This function waits for all answers and retries.
+
+        Args:
+            flow_id_indices (tuple[int]): The flow indentifier
+                indices used to find the given next interface
+                of a load balancer during MDA.
+            extended_classification_flow_id_index (int): a single
+                index to be used for the given option/extension
+                header for evaluating chaining correctness.
+            ttl (int): The radius of the node being evaluated,
+                child of a load balancer.
+
+        Returns:
+            list[Probe]: List of the sent probes.
+        """
+        p = probe.Probe(flow_id_indices, ttl, self.dst_ip, extended_classification_flow_id_index)
+        self.probing.send_probe(p)
+        self.probing.wait()
+        return p
+
+    def load_balancer_node_chaining_correctness_classification(self, ttl: int, load_balancer_node: topology.Node) -> None:
+        """Evaluates the chaining correctness of a node
+
+        Resends the probe used to discover a child interface
+        of a load balancer a given number of times, according to
+        the level of confidence, adding either options (IPv4) or
+        extension headers (IPv6) to evaluate the chaining
+        correctness of the load balancer.
+
+        Args:
+            ttl (int): The radius of the node to be evaluated.
+            load_balancer_node (topology.Node): The load balancer
+                node to be evaluated for chaining correctness.
+
+        Returns:
+            None
+        """
+
+        probes_count = self.nprobes[self.alpha][2]
+        child_node = load_balancer_node.next_hops[0]
+        child_ttl = ttl + 1
+        flow_id_indexes = identifiers.get_discovery_flow_ids(child_ttl, child_node.ip)[0]
```

Comment:
I think there is a misunderstanding going on. It is not the indices that should match. The first step is to get the first next hop, which is stored on `child_node` (line 535 up there). Then, at line 537, I use the `get_discovery_flow_ids` function passing both the TTL and the interface of the next hop. So my understanding is whichever item I use on the row 537 will lead me to that same hop and interface, which is what I want. I use the index 0 for simplicity.

Reply:
Ooops, I missed that `child_node.ip` is being passed to `get_discovery_flow_ids` explicitly. Maybe just add a comment to avoid this in the future. I think the main points is that you get any child in 535, then any flow ID going to *that child* in 537.